src/state/stake/useFarmsPortfolio.ts

```diff
@@ -0,0 +1,212 @@
+  const activeFarms = STAKING[chain]
+
+  const filteredFarms = farmIds ? activeFarms.filter(farm => farmIds?.includes(farm.ID)) : activeFarms
+  const farms = filteredFarms.map(farm => ({
+    farmId: farm.ID,
+    chefVersion: farm.chefVersion,
+    poolId: farm.poolId,
+    rewarderAddress: farm.rewarderAddress,
+    v1args: [farm.poolId.toString(), account],
+    doubleRewardToken: activeFarms[farm.ID].doubleRewardToken
+  }))
+
+  const v1Farms = farms.filter(farm => farm.chefVersion === ChefVersions.V1)
+  const v2Farms = farms.filter(farm => farm.chefVersion === ChefVersions.V2)
+
+  const contractv1 = useMasterChefV2ContractForVersion(ChefVersions.V1)
+  const contractv2 = useMasterChefV2ContractForVersion(ChefVersions.V2)
+
+  const callResultIsLoading = (result: CallState[]) => {
+    return result.some(data => data?.loading === true)
+  }
+  const callResultError = (result: CallState[]) => {
+    return result.some(data => data?.error)
+  }
+
+  // Complex rewards
+
+  const complexRewarderAddressList = v2Farms.map(farm => farm.rewarderAddress).filter(address => address)
+  const complexRewarderContractList = useComplexRewarderMultipleContracts(complexRewarderAddressList)
+
+  const complexRewardsContractAdressList =
+    complexRewarderContractList?.map(contract => (complexRewarderContractList ? contract?.address : undefined)) ?? []
+
+  const pendingComplexRewards = useMultipleContractSingleData(
+    complexRewardsContractAdressList,
+    new Interface(COMPLEX_REWARDER),
+    'pendingTokens',
+    [0, account, '0']
+  )
+
+  const complexRewardsLoading = callResultIsLoading(pendingComplexRewards)
+
+  const mapDoubleRewardToken = (addressToSearch: string) => {
+    const farm = v2Farms.find(farm => farm.doubleRewardToken.address === addressToSearch)
+    return farm?.doubleRewardToken
+  }
+
+  const earnedComplexRewardPool =
+    pendingComplexRewards.length && !complexRewardsLoading
+      ? pendingComplexRewards.map(farmData => ({
+          rewardAmount: JSBI.BigInt(farmData?.result?.rewardAmounts?.[0] ?? 0),
+          rewardToken: mapDoubleRewardToken(farmData?.result?.rewardTokens[0]) ?? dummyToken
+        }))
+      : null
+
+  const complexTokenAmounts = earnedComplexRewardPool?.map(pool => ({
+    token: pool.rewardToken,
+    tokenAddr: pool.rewardToken.address,
+    amount: new TokenAmount(AURORA[ChainId.AURORA], JSBI.BigInt(pool.rewardAmount))
+  }))
+
+  const complexRewardsFarmAmounts: FarmAmount = {}
+
+  complexTokenAmounts?.forEach(tokenReward => {
+    const tokenSymbol = tokenReward.token?.symbol ?? ''
+    return complexRewardsFarmAmounts.hasOwnProperty(tokenSymbol)
+      ? (complexRewardsFarmAmounts[tokenSymbol].amount = complexRewardsFarmAmounts[tokenSymbol].amount.add(
+          tokenReward.amount
+        ))
+      : (complexRewardsFarmAmounts[tokenSymbol] = {
+          token: tokenReward.token,
+          amount: tokenReward.amount,
+          address: tokenReward.tokenAddr
+        })
+  })
```

Comment:
We probably shouldn't return object property assignments

```suggestion
  complexTokenAmounts?.forEach(({ token, amount, tokenAddr }) => {
    const tokenSymbol = token?.symbol ?? ''

    if (complexRewardsFarmAmounts.hasOwnProperty(tokenSymbol)) {
      complexRewardsFarmAmounts[tokenSymbol].amount = complexRewardsFarmAmounts[tokenSymbol].amount.add(amount)
    } else {
      complexRewardsFarmAmounts[tokenSymbol] = { amount, address: tokenAddr, token }
    }
  })
```